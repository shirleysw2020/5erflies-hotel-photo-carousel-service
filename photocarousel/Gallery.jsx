import React, { useState, useEffect } from 'react';
import styled from 'styled-components';
import Button from './Button.jsx';

const Gallery = ({list, openModal, closeModal, show}) => (

  <RoundedCorner>
    <MainWrapper>
      <Imgagery src={list.photos[0]} onClick={(e) => openModal(0)}/>
    </MainWrapper>
    <SubWrapper>
      {list.photos.slice(1, 5).map((photo, index) => {
        return (<Grid picIndex={index}>
            <Imgagery src={photo} key={index} onClick={(e) => openModal(index + 1)}/>
        </Grid>)
      })}
      <Button
        openModal={openModal}
        closeModal={closeModal}
        show={show}
        list={list}
      />
    </SubWrapper>
  </RoundedCorner>
)

const RoundedCorner = styled.div`
  border-radius: 13px;
  overflow: hidden;
  /* height: 372px; */
  display: flex;
`;

const Imgagery = styled.img`
  width: 100%;
  /* height: auto; */
  height: 100%;
  cursor: pointer;
  display: block;
  &: hover {
    -webkit-filter: brightness(85%);
    -webkit-transition: all 0.6s ease;
    transition: all 0.6s ease;
  };
`;

const MainWrapper = styled.div`
  max-width: 1120px;
  flex: 1;
  /* float: left; */
  /* width: 49.3%; */
  margin-right: 0.7%;
`;

const SubWrapper = styled.div`
  display: flex;
  /* width: 50%; */
  flex-flow: wrap;
  position: relative;
  flex: 1;
`;

const Grid = styled.div`
  margin-right: ${props => props.picIndex % 2 === 0 ? '1.3%' : '0'};
  width: 49.35%;
  margin-bottom: ${props => props.picIndex < 2 ? '1.3%' : '0'};
  height: 49%;
  float: left;
  overflow: hidden;
`;


export default Gallery;
