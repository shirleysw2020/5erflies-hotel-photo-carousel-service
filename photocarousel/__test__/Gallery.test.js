// import React from 'react';
// import { shallow } from 'enzyme';
// import Gallery from '../Gallery.jsx';

// //describe is
// describe('Gallery', () => {
//   it('should render correctly with no props', () => {
//     const component = shallow(<Gallery/>);

//     expect(component).toMatchSnapshot();
//   });
// });
// wrapper.instance.function

import React from 'react';
import renderer from 'react-test-renderer';
import Gallery from '../Gallery.jsx';

it('renders correctly when there is one item', () => {
  const list = {
    photos: ["https://5erflies.s3-us-west-1.amazonaws.com/1f57bf4f-0b96-4b38-81ad-afc9f739a7a8.jpg",
    "https://5erflies.s3-us-west-1.amazonaws.com/7fb02bb9-0965-4ffc-a59d-8d7e64f4d9f8.jpg",
    "https://5erflies.s3-us-west-1.amazonaws.com/8e0e456c-10c0-4faa-8297-19f9453937f4.jpg",
    "https://5erflies.s3-us-west-1.amazonaws.com/345b5185-33e6-402b-90d5-93c5d8b7d0e9.jpg",
    "https://5erflies.s3-us-west-1.amazonaws.com/9383d706-87a3-4da1-942e-690b6c438d5c.jpg"]
  }
  const tree = renderer.create(<Gallery list={list} />).toJSON();
  expect(tree).toMatchSnapshot();
});

// //     it('should not render any <button> when operator is not passed in props', () => {
//   expect(wrapper.find('button')).toHaveLength(0);
// });

/*shallow() tests component we provide, and ignores any child components present in the component tree thereafter; if we had a <Header /> and <Footer /> component within <App /> for example, they would be ignored in this test.

In Enzyme we can do such a test with mount(); a full rendering test that takes the entire component tree and lifecycle methods into consideration.
{list: [photo1, photo2, photo3]}
*/