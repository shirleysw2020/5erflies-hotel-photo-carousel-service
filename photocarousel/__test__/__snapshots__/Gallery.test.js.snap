// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there is one item 1`] = `
<div
  className="sc-bwzfXH kWcmvg"
>
  <div
    className="sc-bxivhb dogdys"
  >
    <img
      className="sc-htpNat hcRaaq"
      onClick={[Function]}
      src="https://5erflies.s3-us-west-1.amazonaws.com/1f57bf4f-0b96-4b38-81ad-afc9f739a7a8.jpg"
    />
  </div>
  <div
    className="sc-ifAKCX cHkelG"
  >
    <div
      className="sc-EHOje gozIrZ"
    >
      <img
        className="sc-htpNat hcRaaq"
        onClick={[Function]}
        src="https://5erflies.s3-us-west-1.amazonaws.com/7fb02bb9-0965-4ffc-a59d-8d7e64f4d9f8.jpg"
      />
    </div>
    <div
      className="sc-EHOje jeQsCS"
    >
      <img
        className="sc-htpNat hcRaaq"
        onClick={[Function]}
        src="https://5erflies.s3-us-west-1.amazonaws.com/8e0e456c-10c0-4faa-8297-19f9453937f4.jpg"
      />
    </div>
    <div
      className="sc-EHOje EPdPm"
    >
      <img
        className="sc-htpNat hcRaaq"
        onClick={[Function]}
        src="https://5erflies.s3-us-west-1.amazonaws.com/345b5185-33e6-402b-90d5-93c5d8b7d0e9.jpg"
      />
    </div>
    <div
      className="sc-EHOje fzWQRz"
    >
      <img
        className="sc-htpNat hcRaaq"
        onClick={[Function]}
        src="https://5erflies.s3-us-west-1.amazonaws.com/9383d706-87a3-4da1-942e-690b6c438d5c.jpg"
      />
    </div>
    <button
      className="sc-bdVaJa gwQknd"
    >
      Show all photos
    </button>
  </div>
</div>
`;
